datafusion/core/src/physical_optimizer/hash_build_probe_order.rs

```diff
@@ -174,6 +174,13 @@ impl PhysicalOptimizerRule for HashBuildProbeOrder {
                     *hash_join.partition_mode(),
                     hash_join.null_equals_null(),
                 )?;
+                if matches!(
```

Comment:
I think the same test should apply to `Anti` joins as well, should it not?

Reply:
Although maybe Anti joins don't have their inputs swapped so it can't happen 🤔

Reply:
Indeed. I figured it could be added later when RightAnti is implemented. I will add a note to the ticket.

One thing I also observed is that adding the projection for other types wouldn't be needed if we would update the join columns itself instead of fixing it afterwards, but that requires some more plumbing to get right.